src/test/javascript/spec/util/shared/profile-toggle-hide.directive.spec.ts

```diff
@@ -0,0 +1,105 @@
+import { SecuredImageComponent } from 'app/shared/image/secured-image.component';
+import { TranslateDirective } from 'app/shared/language/translate.directive';
+import { MockComponent, MockDirective, MockModule, MockPipe } from 'ng-mocks';
+import { PieChartModule } from '@swimlane/ngx-charts';
+
+describe('ProfileToggleHideDirective', () => {
+    let fixture: ComponentFixture<CourseCardComponent>;
+    let component: CourseCardComponent;
+
+    let profileToggleService: ProfileToggleService;
+
+    beforeEach(() => {
+        TestBed.configureTestingModule({
+            imports: [ArtemisTestModule, RouterTestingModule.withRoutes([]), MockModule(PieChartModule)],
+            declarations: [
+                CourseCardComponent,
+                MockPipe(ArtemisTranslatePipe),
+                MockPipe(ArtemisTimeAgoPipe),
+                MockRouterLinkDirective,
+                MockComponent(SecuredImageComponent),
+                MockDirective(TranslateDirective),
+                ProfileToggleHideDirective,
+            ],
+            providers: [{ provide: JhiWebsocketService, useClass: MockWebsocketService }],
+        })
+            .compileComponents()
+            .then(() => {
+                fixture = TestBed.createComponent(CourseCardComponent);
+                component = fixture.componentInstance;
+                profileToggleService = TestBed.inject(ProfileToggleService);
+
+                const course = new Course();
+                course.id = 123;
+                component.course = course;
+            });
+    });
+
+    afterEach(() => {
+        jest.restoreAllMocks();
+    });
+
+    it('should always be enabled if decoupling is not enabled', () => {
+        profileToggleService.initializeProfileToggles([]);
+
+        fixture.detectChanges();
+        component.ngOnChanges();
+
+        let lectureButton = fixture.debugElement.query(By.directive(ProfileToggleHideDirective));
+        // Not hidden because 'DECOUPLING' is not activated
+        expect(lectureButton.classes['d-none']).toBeFalsy();
+
+        profileToggleService.initializeProfileToggles([ProfileToggle.LECTURE]);
+
+        fixture.detectChanges();
+        component.ngOnChanges();
+
+        lectureButton = fixture.debugElement.query(By.directive(ProfileToggleHideDirective));
+        // Not hidden because 'DECOUPLING' is not activated - independent of LECTURE
+        expect(lectureButton.classes['d-none']).toBeFalsy();
+    });
+
+    it('should always be disabled if decoupling is enabled', () => {
+        profileToggleService.initializeProfileToggles([ProfileToggle.DECOUPLING]);
+
+        fixture.detectChanges();
+        component.ngOnChanges();
+
+        const lectureButton = fixture.debugElement.query(By.directive(ProfileToggleHideDirective));
+        // Hidden because DECOUPLING is present but LECTURE is not
+        expect(lectureButton.classes['d-none']).toBeTrue();
```

Comment:
```suggestion
        expect(lectureButton.classes['d-none']).not.toBeNull();
```
The component exists rather than being a `true` value.